crates/file_store/src/store.rs

```diff
@@ -25,16 +24,17 @@ impl<C> PersistBackend<C> for Store<C>
 where
     C: Append + serde::Serialize + serde::de::DeserializeOwned,
 {
-    type WriteError = std::io::Error;
-
-    type LoadError = IterError;
-
-    fn write_changes(&mut self, changeset: &C) -> Result<(), Self::WriteError> {
-        self.append_changeset(changeset)
+    fn write_changes(&mut self, changeset: &C) -> anyhow::Result<()> {
+        self.append_changeset(changeset).map_err(|error| {
+            anyhow::Error::msg(error.to_string())
+                .context("Failed to write changes to the persistence backend.")
+        })
```

Comment:
I would use `anyhow` macro.

```suggestion
        self.append_changeset(changeset)
            .map_err(|e| anyhow!(e).context("failed to write changes to persistence backend"))
```